drivers/watchdog/iwdg_stm32.c

```diff
@@ -1,36 +1,77 @@
 /*
  * Copyright (c) 2016 Open-RnD Sp. z o.o.
+ * Copyright (c) 2017 RnDity Sp. z o.o.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/**
- * @brief Driver for Independent Watchdog (IWDG) for STM32 MCUs
- *
- * Based on reference manual:
- *   STM32F101xx, STM32F102xx, STM32F103xx, STM32F105xx and STM32F107xx
- *   advanced ARM ® -based 32-bit MCUs
- *
- * Chapter 19: Independent watchdog (IWDG)
- *
- */
-
 #include <watchdog.h>
 #include <soc.h>
 #include <errno.h>
+#include <assert.h>
 
 #include "iwdg_stm32.h"
 
-#define AS_IWDG(__base_addr) \
-	(struct iwdg_stm32 *)(__base_addr)
+/* Minimal timeout in microseconds. */
+#define IWDG_TIMEOUT_MIN	100
+/* Maximal timeout in microseconds. */
+#define IWDG_TIMEOUT_MAX	26214400
 
-static void iwdg_stm32_enable(struct device *dev)
+#define IS_IWDG_TIMEOUT(__TIMEOUT__)		\
+	(((__TIMEOUT__) >= IWDG_TIMEOUT_MIN) &&	\
+	 ((__TIMEOUT__) <= IWDG_TIMEOUT_MAX))
+
+#if defined(CONFIG_SOC_SERIES_STM32F1X) || defined(CONFIG_SOC_SERIES_STM32F3X)
+#define LSI_FREQ	40000
+#elif defined(CONFIG_SOC_SERIES_STM32F4X) || defined(CONFIG_SOC_SERIES_STM32L4X)
+#define LSI_FREQ	32000
+#endif
+
```

Comment:
Why did you choose to define LSI_FREQ and not to use LSI_VALUE from stm32fxxx_ll_rcc.h?